velox/vector/DecodedVector.cpp

```diff
@@ -374,25 +448,23 @@ BufferPtr copyNullsBuffer(
 } // namespace
 
 DecodedVector::DictionaryWrapping DecodedVector::dictionaryWrapping(
-    const BaseVector& wrapper,
+    memory::MemoryPool& pool,
     vector_size_t size) const {
-  VELOX_CHECK(!isIdentityMapping_);
```

Comment:
do we plan to use dictionaryWrapping() when these conditions are not met?

Edit: Just noticed you added a test for these case, curious why we would want to use this API for these cases as it would end up creating inefficient dictionaries. Is it because we plan to add nulls on top of that?

Also, can you update the method comment to remove the part about `Requires
  /// isIdentityMapping() == false and isConstantMapping() == false.` if we plan to keep this?